function/services/application-mapper/index.js

```diff
@@ -14,29 +14,33 @@ const oracleJsonObject = {
             }
         },
         {
-            ADDRESS_DETAILS: {
-                address_type: 'ICA'
-            }
+            ADDRESS_DETAILS: [
+                {
+                    address_type: 'ICA'
+                }
+            ]
         }
     ]
 };
```

Comment:
You could just create this structure at the end when you return these objects. I.e. forget the inital object, do what you need to do, then 
```
return {
    tables: [
        {
            APPLICATION_FORM: applicationFormJson
        },
        {
            ADDRESS_DETAILS: addressDetailsJson
        }
    ]
};
```
at the end? The benefit being you're not hard coding any multidimensional arrays.

Reply:
This was the only change that I didn't implement.

- We will end up doing some hard coding even with the new approach due to the fact that prefix and section_ref are constants and will need to be setup at the start.
- There are some fields which need to be mapped to both tables which will be annoying if they are not kept under the same object
-  If we go with the new approach we will have to do extra logic in the mapApplicationQuestion to check which json object we should be mapping to

Reply:
I've looked at this in a bit more depth, and I still reckon defining this object structure, then hard coding arrays to navigate it, isn't the way to go. In fact, this one object is the source of all these hard coded arrays. I think it could work as follows: 
```
async function mapApplicationDataToOracleObject(data) {
    const applicationFormJson = {
      prefix: 'U',
      section_ref: 'TEMP'
    }
    const addressDetailsJson =  [
      {
        address_type: 'ICA'
      }
    ]
            
            ...
            
    const applicationQuestion = mapApplicationQuestion(data, applicationFormJson, addressDetailsJson);
    
            ...
            
    return {
      tables: [
        {
            APPLICATION_FORM: applicationFormJson
        },
        {
            ADDRESS_DETAILS: addressDetailsJson
        }
      ] 
    }
};
```
In the above approach you'd need to also modify `mapApplicationQuestion` to accept the two objects instead of the current `oracleJsonObject` but if I'm following correctly, that function just passes the object about and eventually scopes to either the `applicationFormJson` or `addressDetailsJson` object anyway. If you still feel strongly it should be done the current way, lets get a chat as I could just be missing something.